src/config.py

```diff
@@ -0,0 +1,64 @@
+import os
+import pprint
+
+import yaml
+
+
+def load_ci_env(debug):
+    print("> [Info] Gathering env variables")
+    build_info = {
+        "commit_range": os.environ.get("TRAVIS_COMMIT_RANGE", "HEAD...HEAD"),
+        "branch": os.environ.get("TRAVIS_BRANCH", ""),
+        "tag": os.environ.get("TRAVIS_TAG", False),
+        "pull_request": os.environ.get("TRAVIS_PULL_REQUEST", "false"),
+        "travis": os.environ.get("TRAVIS", "false"),
+        "event_type": os.environ.get("TRAVIS_EVENT_TYPE", ""),
+        "docker_reg_username": os.environ.get("DOCKER_USERNAME", ""),
+        "docker_reg_password": os.environ.get("DOCKER_PASSWORD", ""),
+    }
+    if debug:
+        pp = pprint.PrettyPrinter(indent=1)
+        info_debug = build_info
+        info_debug["docker_reg_password"] = "XXX"
+        print(">> CI environment configuration: ")
+        pp.pprint(info_debug)
+        print('\n')
+    return build_info
+
+
+def load_image_config(image_type, version):
+    config_path = "config.yml"
+    config = yaml.load(open(config_path))
+
+    # Raise exceptions if a key is not found
+    if image_type not in config:
+        raise KeyError("No configuration is set for this image - Image: " + image_type)
+    if version not in config[image_type]:
+        existing_version = [v for v,_ in config[image_type].items()]
+        raise KeyError("This version is not defined for this image - Defined version: " + ', '.join(existing_version))
```

Comment:
`versions` instead of `version`, both in the variable name and in the message. I'd also add `image_type` to the message, instead of saying "this image".

For a more fluent discussion, please don't resolve the discussions yourself, leave it to the person who reported it in the first place.

Reply:
Got it ! Change done 👍